add.php

```diff
@@ -0,0 +1,102 @@
+<?php
+    ini_set('error_reporting', E_ALL);
+    ini_set('display_errors', 1);
+    ini_set('display_startup_errors', 1);
+
+    // include files
+    require_once('helpers/helpers.php');
+    require_once('helpers/formatters.php');
+    require_once('helpers/fetchers.php');
+    require_once('helpers/formValidation.php');
+
+    // setup default timezone
+    date_default_timezone_set('Europe/Madrid');
+
+    $isAuth = rand(0, 1);
+
+    $userName = 'Katia Sheleh';
+
+    $categories = fetchCategories(); // src => helpers/fetchers.php
+
+    $requiredFields = ['lot-name', 'category', 'message', 'lot-img', 'lot-rate', 'lot-step', 'lot-date'];
+    $errors = [];
+    $rules = [
+        'lot-name' => function () {
+            return validateText($_POST['lot-name'], 10, 70, 'Enter the lot name');
+        },
+        'category' => function () {
+            return validateFilled($_POST['category'], 'Select a category');
+        },
+        'message' => function () {
+            return validateText($_POST['message'], 20, 300, 'Enter the lot description');
+        },
+        'lot-img' => function () {
+            return validateImage('lot-img', __DIR__ . '/uploads/');
+        },
+        'lot-rate' => function () {
+            return validateNotNegativeNumber($_POST['lot-rate'], 'Enter the starting price');
+        },
+        'lot-step' => function () {
+            return validateNotNegativeNumber($_POST['lot-step'], 'Enter the rate step');
+        },
+        'lot-date' => function () {
+            return validateDate($_POST['lot-date']);
+        }
+    ];
+
+
+    // check errors in all fields ($_POST)
+    foreach ($_POST as $key => $value) {
+        if (isset($rules[$key])) {
+            $rule = $rules[$key];
+            $errors[$key] = $rule();
+        }
+    }
+
+    // check errors in the file field ($_FILES)
+    if (isset($_POST['submit'])) {
+        if (isset($rules['lot-img'])) {
+            $rule = $rules['lot-img'];
+            $errors['lot-img'] = $rule();
+        }
+    }
```

Comment:
Для файла нужно отдельно проработать сценарий.

```suggestion
     if (isset($_FILES['lot-img'])) {
        $error = validateImage('lot-img');

        if (null === $error) {
            $filePath = __DIR__ . '/uploads/';
            move_uploaded_file($_FILES['lot-img']['tmp_name'], $filePath . $_FILES['lot-img']['name']);
        } else {
            $errors['lot-img'] = $error;
        }
    }
```